### PR TITLE
Added assert macros SM_ASSERT_FINITE and SM_ASSERT_NOTNAN for conveni…

### PR DESCRIPTION
**Pull request by uschwes**
…ence

### CODE REVIEW

File: sm_common/include/sm/assert_macros.hpp [hunk 1]
```diff
@@ -232,6 +232,21 @@ namespace sm {
       sm::detail::sm_throw_exception<exceptionType>("[" #exceptionType "] ", __FUNCTION__,__FILE__,__LINE__,sm_assert_stringstream.str()); \
     }
 
+#define SM_ASSERT_FINITE(exceptionType, value, message) \
+  if(!(isfinite((value))))           \
```
Comment (HannesSommer) [inline sm_common/include/sm/assert_macros.hpp]: isn't here `std::` missing?

Comment (uschwes) [inline sm_common/include/sm/assert_macros.hpp]: since math.h is included, I don't think so.

Comment (HannesSommer) [inline sm_common/include/sm/assert_macros.hpp]: Ah, okay. Yes, then you are right. Probably one should switch at some point. But find for now.

**Comment by HannesSommer** [comment]
lgtm

**Pull request merged**